### DISCUSSION TITLE
How to set up an interceptor

### DISCUSSION THREAD

**Discussion by donywan** [serverpod/serverpod#2680] [Q&A] [answered]
Most endpoints require login, but only a few do not. How to intercept all endpoints and require login?
```
class MyEndpoint extends Endpoint {
  @override
  bool get requireLogin => true;

  @override
  Set<Scope> get requiredScopes => {Scope.admin};

  Future<void> myMethod(Session session) async {
    ...
  }
  ...
}
```

**Comment by donywan** [comment]
Anyone? I don't want to write `bool get requireLogin => true;` for every endpoint.

**Comment by hampuslavin** [comment, marked as answer]
Hi @donywan!
Currently there is no way to solve the problem you're describing. However, we are aware of it and we have an open issue with a suggested solution for it (see https://github.com/serverpod/serverpod/issues/912). Feel free to upvote that issue, as that helps us prioritize what features to implement!

**Reply by donywan** [reply]
Ok,come on!
